ast/ast.go

```diff
@@ -263,12 +263,15 @@ type (
 	// large time ranges are processed and streamed efficiently.
 	// The limit parameter specifies the number of different groups that can
 	// be aggregated over. When absent, the runtime defaults to an appropriate value.
+	//
+	// xxx comment sorted
 	GroupByProc struct {
 		Node
 		Duration Duration     `json:"duration"`
 		Limit    int          `json:"limit,omitempty"`
 		Keys     []Assignment `json:"keys"`
 		Reducers []Reducer    `json:"reducers"`
+		Sorted   int          `json:"sorted"`
```

Comment:
```suggestion
		Sorted   int          `json:"sorted,omitempty"`
```

Reply:
Also, Sorted is a pretty vague name, so I think we should call this InputSortDir/input_sort_dir.

Reply:
👍

Reply:
More importantly, do you think the zql flag `-sorted` should also be renamed?

Reply:
I don't think we should expose this through ZQL.

Instead, I think we should add a `sortKey string` parameter to driver.Compile*, set it appropriately when calling those functions (to `"ts"` in zqd; to `""` in zq for now, ultimately wiring it to a command line flag), and then use it as input to an analysis that will set ast.GroupByProc.InputSortDir to the correct value before calling proc.CompileProc.

That's obviously more work, but we don't need to do a thorough job of it now. We just need enough to catch the Brim histogram AST, which seems pretty straightforward.